dev/com.ibm.jbatch.container/src/com/ibm/jbatch/container/services/impl/JPAPersistenceManagerImpl.java

```diff
@@ -2685,34 +2698,169 @@ private int getJobInstanceTableVersion(PersistenceServiceUnit psu) throws Except
-        } catch (javax.persistence.PersistenceException e) {
-            Throwable cause = e.getCause();
+        } catch (javax.persistence.PersistenceException e3) {
+            Throwable cause = e3.getCause();
             while (cause != null) {
                 if (cause instanceof SQLSyntaxErrorException &&
                     cause.getMessage() != null &&
-                    cause.getMessage().contains("UPDATETIME")) {
+                    cause.getMessage().contains("GROUPNAMES")) {
                     // The column isn't there.
-                    logger.fine("The UPDATETIME column does not exist, job instance table version = 1");
-                    instanceVersion = 1;
-                    return instanceVersion;
+                    try {
+                        Integer exec = new TranRequest<Integer>(em) {
+                            @Override
+                            public Integer call() throws Exception {
+                                // Verify that UPDATETIME column exists by running a query against it.
+                                String queryString = "SELECT COUNT(x.lastUpdatedTime) FROM JobInstanceEntityV2 x";
+                                TypedQuery<Long> query = em.createQuery(queryString, Long.class);
+                                query.getSingleResult();
+                                logger.fine("The UPDATETIME column exists, job instance table version = 2");
+                                instanceVersion = 2;
+                                return instanceVersion;
+                            }
+                        }.runInNewOrExistingGlobalTran();
+
+                        return exec;
+                    } catch (javax.persistence.PersistenceException e2) {
+                        cause = e2.getCause();
+                        while (cause != null) {
+                            if (cause instanceof SQLSyntaxErrorException &&
+                                cause.getMessage() != null &&
+                                cause.getMessage().contains("UPDATETIME")) {
+                                // The column isn't there.
+                                logger.fine("The UPDATETIME column does not exist, job instance table version = 1");
+                                instanceVersion = 1;
+                                return instanceVersion;
+                            }
+                            cause = cause.getCause();
+                        }
+                    }
                 }
-                cause = cause.getCause();
             }
             logger.fine("Unexpected exception while checking job instance table version, re-throwing");
-            throw e;
+            throw e3;
         } finally {
             em.close();
         }
     }
+
+    @Override
+    public List<String> getGroupNamesForJobID(long jobInstanceID) throws NoSuchJobInstanceException {
+        EntityManager em = getPsu().createEntityManager();
+        try {
```

Comment:
All JPA should now be done inside a global tran !

Reply:
should look like this:

Integer exec = new TranRequest<Integer>(em) {
                @Override
                public Integer call() throws Exception {
                    <JPA CODE>
                }
            }.runInNewOrExistingGlobalTran();